### PR TITLE
tl-10. tests with DTO pattern and factory methods, to verify response…

### PR DESCRIPTION
**Pull request by Marusia2**
… codes and behavior

### CODE REVIEW

File: tests/dto/order-dto-hw.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+export class OrderDtoHw {
```
Comment (tallinn-learning) [inline tests/dto/order-dto-hw.ts]: let;s keep names meaningful 
OrderDtoHw is not clear

Comment (tallinn-learning) [reply tests/dto/order-dto-hw.ts]: so this is not order but LoanApplication

File: tests/dto/order-dto-hw.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      Math.floor(Math.random() * 10),
+      Math.floor(Math.random() * 50) + 17,
+      true,
+      Math.floor(Math.random() * 1000),
+      //[6, 9, 12][Math.floor(Math.random() * 3)],
+      [9][Math.floor(Math.random())],
+    )
+  }
+  // method to return High risk with random data
+  static calculateHighRiskScoreBasedOnIncomeWithValidData(): OrderDtoHw {
```
Comment (tallinn-learning) [inline tests/dto/order-dto-hw.ts]: calculateHighRiskScoreBasedOnIncomeWithValidData
generateApplicationWithHighRiskScoreBasedOnIncomeWithValidData

File: tests/readme.md [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+|   5    | PUT request. Update an order by providing correct data should receive code 200                                        | PASS   |
+|   6    | PUT request. Update an order by providing correct data but for the unauthorized user should receive code 401          | PASS   |
+|   7    | PUT request. Update an order with incorrect ID but the for the authorized user should receive code 400                | PASS   |
+|   8    | DELETE request. Delete an order by providing correct data should receive code 204                                     | PASS   |
+|   9    | DELETE request. Delete order by providing an invalid order ID should receive code 400                                 | PASS   |
+|   10   | GET request. Login with a valid username and password should receive code 200 and API key                             | PASS   |
+|   11   | GET request. Login with missing username and password data should receive code 500                                    | PASS   |
+|   12   | POST request. Calculate LOW RISK score with valid random data should return code 200                                  | PASS   |
```
Comment (tallinn-learning) [inline tests/readme.md]: let's split topics for order & loan

File: tests/simple-solution-api.spec.ts [hunk 4]
```diff
@@ -184,3 +183,121 @@ test('Get order with missing data should receive code 500', async ({ request })
   console.log('response body:', await response.json())
   expect(response.status()).toBe(StatusCodes.INTERNAL_SERVER_ERROR)
 })
+
+// 10 Homework
+//10.1  Scenario to check Low Risk score
+test('post request calculate Low risk score with valid data returns code 200', async ({
+  request,
+}) => {
+  const orderDtoHw = OrderDtoHw.calculateLowRiskScoreBasedOnIncomeWithValidData()
```
Comment (tallinn-learning) [inline tests/simple-solution-api.spec.ts]: pls rename objects here as well

**Comment by Marusia2** [comment]
@tallinn-learning  Fixed according the comments:)

File: tests/simple-solution-api.spec.ts [hunk 5]
```diff
@@ -184,3 +183,127 @@ test('Get order with missing data should receive code 500', async ({ request })
   console.log('response body:', await response.json())
   expect(response.status()).toBe(StatusCodes.INTERNAL_SERVER_ERROR)
 })
+
+// 10 Homework
+//10.1  Scenario to check Low Risk score
+test('post request calculate Low risk score with valid data returns code 200', async ({
+  request,
+}) => {
```
Comment (vladimir-tl) [inline tests/simple-solution-api.spec.ts]: better to create a separate class and put loan tests there
this is last comment

**Comment by Marusia2** [comment]
@tallinn-learning thank you for comment,  fixed that :)  created a separate file to run loan tests;)))

**Pull request merged**